cmd/collectors/zapi/collector/zapi.go

```diff
@@ -405,6 +412,9 @@ func (z *Zapi) PollData() (map[string]*matrix.Matrix, error) {
 	z.AddCollectCount(count)
 
 	if numInstances == 0 {
+		if z.RunPluginsIfNoData {
```

Comment:
It's generally a bad idea to return a value and error. The typical pattern is nil, err or something, nil. It invites nil panics to mix the two. 

Do we need to return the error? If we return an ErrNoInstance won't the collector be moved to standby in `collector.go`? If you agree that the error is not needed, then we can revert the changes to `collector.go` too.

Would this work?
```go
if numInstances == 0 && !z.RunPluginsIfNoData {
	return nil, errs.New(errs.ErrNoInstance, "")
}
```

Reply:
added one small function in util for this to use in both rest/zapi collector.